atomic_reactor/plugins/input_osv3.py

```diff
@@ -186,3 +187,68 @@ def run(self):
     @classmethod
     def is_autousable(cls):
         return 'BUILD' in os.environ and 'SOURCE_URI' in os.environ and 'OUTPUT_IMAGE' in os.environ
+
+
+class OSv3SourceContainerInputPlugin(InputPlugin):
+    """Input plugin for building source container images"""
+    key = "osv3_source_container"
+
+    def __init__(self, **kwargs):
+        """
+        constructor
+        """
+        # call parent constructor
+        super(OSv3SourceContainerInputPlugin, self).__init__(**kwargs)
+        self.target_registry = None
+        self.reactor_env = None
+        self.plugins_json = None
+
+    def run(self):
+        build_json = get_build_json()
+        image = os.environ['OUTPUT_IMAGE']
+        self.target_registry = os.environ.get('OUTPUT_REGISTRY', None)
+
+        arrangement_version = None
+        try:
+            user_params = os.environ['USER_PARAMS']
+            user_data = json.loads(user_params)  # TODO validate user params against schema
+            arrangement_version = user_data.get('arrangement_version', None)
+            self.plugins_json = get_plugins_with_user_data(user_params, user_data)
+            # if we get the USER_PARAMS, we'd better get the REACTOR_CONFIG too
+            reactor_config_map = os.environ['REACTOR_CONFIG']
+            self.reactor_env = read_yaml(reactor_config_map, 'schemas/config.json')
+        except KeyError:
+            try:
+                self.plugins_json = os.environ['ATOMIC_REACTOR_PLUGINS']
+            except KeyError:
+                raise RuntimeError("No plugin configuration found!")
+
+        if arrangement_version and arrangement_version <= 5:
+            raise ValueError('arrangement_version <= 5 is no longer supported')
+
+        self.plugins_json = json.loads(self.plugins_json)
+        # validate json before performing any changes
+        read_yaml(json.dumps(self.plugins_json), 'schemas/plugins.json')
+
+        input_json = {
+            'image': image,
+            'openshift_build_selflink': build_json.get('metadata', {}).get('selfLink', None)
+        }
+        input_json.update(self.plugins_json)
+
+        self.log.debug("build json: %s", input_json)
+
+        # TODO remove koji plugins when koji is not defined
+
+        read_yaml(json.dumps(self.plugins_json), 'schemas/plugins.json')
+
+        return input_json
+
+    @classmethod
+    def is_autousable(cls):
+        return (
+            'OUTPUT_IMAGE' in os.environ and
+            'USER_PARAMS' in os.environ and
+            json.loads(os.environ['USER_PARAMS'])
+                .get('kind', '__unknown__') == 'source_containers_user_params'  # TODO use osbs-client constant
```

Comment:
:eyes: 

I'll try to create that PR soon in osbs-client, but it shouldn't prevent this PR from merging